test/token/add_token_test.dart

```diff
@@ -34,6 +34,52 @@ void main() {
     }).validate();
   });
 
+  group('with environment variable creates tokens.json that contains env var',
+      () {
+    test('without environment variable provided', () async {
+      await d.tokensFile({
+        'version': 1,
+        'hosted': [
+          {'url': 'https://example.com', 'token': 'abc'},
+        ]
+      }).create();
+
+      await runPub(
+        args: ['token', 'add', 'https://example.com/', '--env-var', 'TOKEN'],
+        error: 'Environment variable TOKEN is not defined.',
+      );
+
+      await d.tokensFile({
+        'version': 1,
+        'hosted': [
+          {'url': 'https://example.com', 'env': 'TOKEN'},
+        ]
+      }).validate();
+    });
+
+    test('with environment variable provided', () async {
+      await d.tokensFile({
+        'version': 1,
+        'hosted': [
+          {'url': 'https://example.com', 'token': 'abc'},
+        ]
+      }).create();
+
+      await runPub(
+        args: ['token', 'add', 'https://example.com/', '--env-var', 'TOKEN'],
+        environment: {'TOKEN': 'secret'},
+        error: isNot(contains('Environment variable TOKEN is not defined.')),
+      );
+
+      await d.tokensFile({
+        'version': 1,
+        'hosted': [
+          {'url': 'https://example.com', 'env': 'TOKEN'},
+        ]
+      }).validate();
+    });
```

Comment:
We would also need a test that the credential is actually read from the envvar.

Reply:
I'll work on tests with mocked server on another PR.